net/initiate_test.go

```diff
@@ -1,14 +1,35 @@
 package net
 
 import (
+	"encoding/hex"
 	"testing"
 	"time"
 
+	"github.com/cockroachdb/apd/v3"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+
 	"github.com/athanorlabs/atomic-swap/common/types"
+	"github.com/athanorlabs/atomic-swap/net/message"
+	pcommon "github.com/athanorlabs/atomic-swap/protocol"
 
 	"github.com/stretchr/testify/require"
 )
 
+func createSendKeysMessage(t *testing.T) *message.SendKeysMessage {
+	keysAndProof, err := pcommon.GenerateKeysAndProof()
+	require.NoError(t, err)
+	return &message.SendKeysMessage{
+		OfferID:            types.Hash{},
+		ProvidedAmount:     new(apd.Decimal),
+		PublicSpendKey:     keysAndProof.PublicKeyPair.SpendKey(),
+		PublicViewKey:      keysAndProof.PublicKeyPair.ViewKey(),
+		PrivateViewKey:     keysAndProof.PrivateKeyPair.ViewKey(),
+		DLEqProof:          hex.EncodeToString(keysAndProof.DLEqProof.Proof()),
+		Secp256k1PublicKey: keysAndProof.Secp256k1PublicKey,
+		EthAddress:         ethcommon.Address{},
```

Comment:
Tests depend on the `OfferID` and the `ETHAddress` being all zeros.  I'll address that in a follow-up PR.